internal/common/context.bzl

```diff
@@ -0,0 +1,328 @@
+            transitive_sources.append(dep_js_library.all_sources)
+            transitive_dep_modules.append(dep_js_library.all_dep_modules)
+        elif JsModuleInfo in dep:
+            dep_js_module = dep[JsModuleInfo]
+            transitive_dep_modules.append(dep_js_module.all_dep_modules)
+            transitive_dep_module_targets.append(dep_js_module.all_dep_module_targets)
+        elif hasattr(dep, "tags") and "NODE_MODULES_MARKER" in getattr(dep, "tags"):
+            # The dependency is a module defined by rules_nodejs
+            direct_modules += dep
+
+        if JsModuleInfo in dep:
+            dep_js_module = dep[JsModuleInfo]
+            dep_module_targets.append(dep_js_module.module_root)
+            dep_modules.append(dep_js_module)
+
+        if NpmPackagesInfo in dep:
+            # The dependency is a node_modules directory installed by npm_packages
+            node_modules_dirs.append(dep[NpmPackagesInfo].installed_dir)
+
+    all_src_files = depset(
+        direct = src_files,
+        transitive = transitive_sources,
+    )
+
+    all_dep_modules = depset(
+        direct = dep_modules,
+        transitive = transitive_dep_modules,
+    )
+
+    all_dep_module_targets = depset(
+        direct = dep_module_targets,
+        transitive = transitive_dep_module_targets,
+    )
+
+    return JsLibraryInfo(
+        js = js,
+        package_path = js.package_path,
+        srcs = src_files,
+        all_sources = all_src_files,
+        node_modules_dirs = node_modules_dirs,
+        dep_modules = dep_modules,
+        all_dep_modules = all_dep_modules,
+        dep_module_targets = dep_module_targets,
+        all_dep_module_targets = all_dep_module_targets,
+    )
+
+def _library_to_source_info(js, library, gen_scripts = None):
+    """Create a JsSource provider for a given library
+    The library is a target, but this resolves the actual source files needed
+    to build the js library.
+    """
+
+    return JsSourceInfo(
+        js = js,
+        srcs = library.all_sources,
+        gen_scripts = gen_scripts,
+        library = library,
+    )
+
+def _library_to_module_info(js, library, module_root, module_name):
+    return JsModuleInfo(
+        js = js,
+        workspace_name = js.workspace_name,
+        all_dep_modules = library.all_dep_modules,
+        all_dep_module_targets = library.all_dep_module_targets,
+        module_root = module_root,
+        module_name = module_name,
+    )
+
+def _script_args(js, script_file):
+    """Create Args object that can be used with js.run_js()
+
+    Args:
+    js: JsContext object
+    script_file: File object for the script to be run
+    """
+    if script_file == None:
+        fail("No script file provided to script args")
+
+    args = js.actions.args()
+
+    # If the args get too big then spill over into the param file
+    args.use_param_file("--param=%s")
+    args.set_param_file_format("multiline")
+
+    # args.add(script_file)
```

Comment:
Shouldn't this be added back?

Reply:
I don't think so, the `js.run_js` action takes the script in as a parameter. Also, because [Args](https://docs.bazel.build/versions/master/skylark/lib/Args.html) is backed by a depset, the order isn't guaranteed. So `js.run_js` makes sure that the script argument is added before the script args.

Reply:
Makes sense. In that case why not remove the second argument to this function, and stop providing it in each call to `script_args`?

Reply:
Fixed.